### DISCUSSION TITLE
Referring to field defined at outer most level at some inner level where duplicated field is present

### DISCUSSION THREAD

**Discussion by suimong** [tweag/nickel#1777] [Q&A] [answered]
Consider the following

```nickel
# sample.ncl
{
  foo = 1,
  bar = {
    foo = 5,
  },
  baz = {
    foo = 3,
    cux = foo + 1,
  },

}
```
Now `baz.cux` evaluates to 4 which is expected, and as long as some `foo` is nested within another field, there's no problem referring to it (e.g. `cux = bar.foo +1` evaluates to 6), but what if I want to refer to the outermost `foo` at `baz.cux`? I couldn't seem to find a way.

**Comment by yannham** [comment, marked as answer]
The simplest work around currently is to capture the external variable in a let binding before it's shadowed:

```nickel
# sample.ncl
{
  foo = 1,
  bar = {
    foo = 5,
  },
  baz = let bar_foo = bar.foo in {
    foo = 3,
    cux = foo + bar_foo + 1,
  },
}
```

It can sometimes become a bit verbose. Possible future solutions could include special keywords like `self`/`super`. Alternatively, a `nonrec` keyword to specify that we want some records to _not_ be recursive could also be a solution.

**Reply by suimong** [reply]
Another trick learned, thank you! Though for reference to future me or others stumbled upon this, what @yannham mean is:

```nickel
# sample.ncl
{
  foo = 1,
  bar = {
    foo = 5,
  },
  baz = let outer_foo = foo in {  
    foo = 3,
    cux = foo + outer_foo + 1,  # evaluates to 5
  },
}
```
As for future solutions, I like the idea of `self`/`super`, but obviously it requires some thinking about its proper semantic and syntactic meaning, as well as the cost of introducing another keyword.
